colosseum/fonts.py

```diff
@@ -0,0 +1,109 @@
+"""Font utilities."""
+
+import os
+import sys
+
+from .exceptions import ValidationError
+
+# Constants
+_GTK_WINDOW = None
+
+
+class FontDatabase:
+    """
+    Provide information about the fonts available in the underlying system.
+    """
+    _FONTS_CACHE = {}
+
+    @classmethod
+    def validate_font_family(cls, value):
+        """Validate a font family with the system found fonts."""
+        if value in cls._FONTS_CACHE:
+            return value
+        else:
+            if check_font_family(value):
+                # TODO: to be filled with a font properties instance
+                cls._FONTS_CACHE[value] = None
+                return value
+
+        raise ValidationError('Font family "{value}" not found on system!'.format(value=value))
+
+
+def _check_font_family_mac(value):
+    """List available font family names on mac."""
+    from ctypes import cdll, util
+    from rubicon.objc import ObjCClass
+    appkit = cdll.LoadLibrary(util.find_library('AppKit'))  # noqa
+    NSFontManager = ObjCClass("NSFontManager")
+    NSFontManager.declare_class_property('sharedFontManager')
+    NSFontManager.declare_property("availableFontFamilies")
+    manager = NSFontManager.sharedFontManager
+    for item in manager.availableFontFamilies:
+        font_name = str(item)
+        if font_name == value:
+            return True
+
+    return False
+
+
+def _check_font_family_linux(value):
+    """List available font family names on linux."""
+    import gi  # noqa
+    gi.require_version("Gtk", "3.0")
```

Comment:
This was what I found on google using pango. I guess this is more in line with what you had in mind?